Libraries/Game/game.cpp

```diff
@@ -0,0 +1,203 @@
+#include "game.h"
+
+/**
+ * @global {Point} gameFieldSize - a variable, which stores the dimensions of the game field
+ */
+Point gameFieldSize;
+
+/**
+ * @function Initializes game
+ * @param {int} size - size of the game field(on the edges the boundaries of the game are situated)
+ * @param {int} sp - speed of the game(1 / refresh rate) in milliseconds
+ * @return {bool} mark of successful initialization
+ */
+bool Game::init(int size, int sp)
+{
+    gameFieldSize.x = size;
+    gameFieldSize.y = size/2;
+    setLabyrinth(gameFieldSize);
+    setSpeed(sp);
+    return true;
+}
+
+/**
+ * @function Main game loop
+ * @return {int} the exit code
+ */
+int Game::run()
+{
+
+    initscr();
+
+    noecho();
+    nodelay(stdscr, TRUE);
+    set_escdelay(0);
+
+    Point screenSize;
+    getmaxyx(stdscr, screenSize.y, screenSize.x);
+
+    Point p, g;
+    p.x = 1;
+    p.y = 1;
+    int dir = 1, len = 5;
+    initSnake(p, dir, len);
+
+    curs_set(0);
+    keypad(stdscr, true); 
+    bool flag = false;
+
+    Ball ball;
+    ball.init(gameFieldSize);
+    ball.generateBall(labyrinth);
+
+    Point* change[2];
+    change[0] = new Point [1];
+    change[1] = new Point [1];
+
+    std::list<Point> currBody;
+    while ( !flag )
+    {
+
+        int command = getch();
+
+        switch (command)
+        {
+            case ERR:
+                break;
+            case KEY_UP:
+                snake.setDirection(MVUP);
+                break;
+            case KEY_DOWN:
+                snake.setDirection(MVDOWN);
+                break;
+            case KEY_LEFT:
+                snake.setDirection(MVLEFT);
+                break;
+            case KEY_RIGHT:
+                snake.setDirection(MVRIGHT);
+                break;
+            case 'q':
+                flag = true;
+                break;
+        }
+        if (flag)
+        {
+            break;
+        }
+        flag = snake.move(labyrinth, &ball, change);
+        updateLabyrinth(change, 1);
+        clear();
+        displayLabyrinth();
+        snake.getCoords(&currBody);
+        for(auto it = currBody.begin(); it != currBody.end(); it++)
```

Comment:
It's better not to redraw all snake (at Windows console this algorithm will produce blinking), just erase last cell and draw new first